sirepo/template/cloudmc.py

```diff
@@ -546,14 +547,17 @@ def _generate_parameters_file(data, run_dir=None):
             cores = 1 if r == "sequential" else sirepo.mpi.cfg().cores
             v.runCommand = f"openmc.run(threads={ cores })"
 
-    v.materials = _generate_materials(data)
-    v.sources = _generate_sources(data)
+    v.incomplete_data_msg = ""
+    v.materials = _generate_materials(data, v)
+    v.sources = _generate_sources(data, v)
     v.sourceFile = _source_filename(data)
     v.maxSampleSourceParticles = SCHEMA.model.openmcAnimation.numSampleSourceParticles[
         5
     ]
-    v.tallies = _generate_tallies(data)
+    v.tallies = _generate_tallies(data, v)
     v.hasGraveyard = _has_graveyard(data)
+    if v.incomplete_data_msg:
+        return f'raise AssertionError("{"Unable to generate sim:" + v.incomplete_data_msg}")'
```

Comment:
You don't want to do string concatenation within `{}` in an f-string. I think this is the same but with less quoting.

```
f'raise AssertionError("Unable to generate sim: {v.incomplete_data_msg}")'
```